docs/dogstatsd/configuration.md

```diff
@@ -0,0 +1,65 @@
+## `dogstatsd_buffer_size`
+
+The `dogstatsd_buffer_size` parameter is indicating two things in Dogstatsd:
+
+* how many bytes must be read each time the socket is read
+* how many bytes maximum the PacketAssembler should put into one packet
+
+Dogstatsd doesn't support reading malformed or incomplete packets from the network, meaning
```

Comment:
It's the reading limitation. We're reading fixed chunk of bytes (`dogstatsd_buffer_size`) on the socket, if the client doesn't take care of sending packets with a size lower than that, we start reading incomplete metrics because the chunk of bytes may not contain the whole metric.

With `dogstatsd_buffer_size` set to 25 and if we receive this metric:

```
"my_metric:60|g|#shell\nsecond2:50|c|#shell"   // 42 chars, \n at the 23rd position
```

`my_metric` will be valid and `second` wont, only `my_metric` will make it to the aggregator.